### PR TITLE
Fix #4421: Add `-C DIR` flag

### LINKED ISSUE #4421: add support for `-C` flag to chdir before running
Supported by big Go since 1.20 for all commands. It's helpful for scripting.  Expected behavior:

- cd's to argument
- runs rest of command

    tinygo run -C path/to/dir some_file.go   # some_file.go is in path/to/dir

See `go help build`:

    % go help build
    usage: go build [-o output] [build flags] [packages]

    [...]

      -C dir
        Change to dir before running the command.
        Any files named on the command line are interpreted after
        changing directories.
        If used, this flag must be the first one in the command line.

    [...]

### PR DESCRIPTION
**Pull request by archie2x**
See #4421 

Implementation is copied from BigGo 1.22.5 with minor modification.

https://github.com/golang/go/blob/d79c350916c637de911d93af689a5e4e7ab5a5bb/src/cmd/go/main.go#L330

### CODE REVIEW

**Comment by deadprogram** [comment]
Hello @archie2x thanks for the PR. Just wondering if you could add a test of some kind, perhaps?

**Comment by archie2x** [comment]
> Hello @archie2x thanks for the PR. Just wondering if you could add a test of some kind, perhaps?

Absolutely. A test didn't originally seem obvious to me but I think something as simple as `tinygo build -C testdata testprogram.go` should work. I'll put that together. Should I rebase?

**Comment by deadprogram** [comment]
> Should I rebase?

Yes, please!

**Comment by archie2x** [comment]
> > Hello @archie2x thanks for the PR. Just wondering if you could add a test of some kind, perhaps?
> 
> Absolutely. A test didn't originally seem obvious to me but I think something as simple as `tinygo build -C testdata testprogram.go` should work. I'll put that together. Should I rebase?

Please let me know if the attached tests suffice for now. I wasn't familiar with the test framework here. I think there may be some opportunity to cleanup how temp directories are used? I'm getting quite a few in my /tmp when I try to run everything locally.

File: tests/testing/chdir/chdir.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	expectDir := os.Getenv("EXPECT_DIR")
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if runtime.GOOS == "windows" {
+		strings.Replace(cwd, "\\", "/", -1)
+	}
+	if cwd != expectDir {
+		log.Fatal("expected:\"%v\" != os.Getwd():\"%v\"", expectDir, cwd)
```
Comment (dgryski) [inline tests/testing/chdir/chdir.go]: `log.Fatalf`

File: tests/testing/chdir/chdir.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+argument.
+*/
+func main() {
+	expectDir := os.Getenv("EXPECT_DIR")
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if runtime.GOOS == "windows" {
+		strings.Replace(cwd, "\\", "/", -1)
```
Comment (dgryski) [inline tests/testing/chdir/chdir.go]: `strings.ReplaceAll` (or even better `filepath.ToSlash`) ?

File: tests/testing/chdir/chdir.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+/*
+Test that this program is 'run' in expected directory. 'run' with expected
+working-directory in 'EXPECT_DIR' environment variable' with{,out} a -C
+argument.
+*/
+func main() {
+	expectDir := os.Getenv("EXPECT_DIR")
+	cwd, err := os.Getwd()
+	if err != nil {
```
Comment (dgryski) [inline tests/testing/chdir/chdir.go:18]: Note this test failed on Windows CI, so we might need more here.

**Review by dgryski — CHANGES_REQUESTED** [review]

File: tests/testing/chdir/chdir.go (hunk 1 above)
Comment (archie2x) [reply tests/testing/chdir/chdir.go]: Fixed

File: tests/testing/chdir/chdir.go (hunk 2 above)
Comment (archie2x) [reply tests/testing/chdir/chdir.go]: Fixed

File: tests/testing/chdir/chdir.go (hunk 3 above)
Comment (archie2x) [reply tests/testing/chdir/chdir.go:18]: Fixed, thanks for the review!

**Review by dgryski — APPROVED** [review]
LGTM. CI passes on Windows now.

**Comment by deadprogram** [comment]
Thank you very much for the PR @archie2x and to @dgryski for review.

Now squash/merging.

**Pull request merged**